src/Timestamp.js

```diff
@@ -5,6 +5,8 @@ import Long from "long";
  * @typedef {import("@hashgraph/proto").ITimestamp} proto.ITimestamp
  */
 
+const MAX_NS = Long.fromNumber(999999999);
```

Comment:
You have an off by one error. If the timestamp is sec=0,ns=999999998, incrementing by one nano should give you sec=0,ns=999999999 but it gives you sec=1,ns=0. Please add a test so you can verify.

Reply:
You're right, fixed and added regression test.